File: kotlin/series/src/main/kotlin/Series.kt
object Series {
    fun slices(n: Int, input: String) =
        if (n < 1 || input.isBlank() || n > input.length)
            throw IllegalArgumentException("n must be a positive non-zero integer that is not larger than the length of the input string, and input must be a non-empty string")
        else input.map(Character::getNumericValue).windowed(n)
}

File: fsharp/series/Series.fs
module Series

open System

let slices (s: string) len =
    if len < 1 || s.Length < len then
        None
    else
        s
        |> Seq.windowed len
        |> Seq.map (Seq.toArray >> String)
        |> Seq.toList
        |> Some
